app/src/androidTest/java/com/google/jetpackcamera/ImageCaptureDeviceTest.kt

```diff
@@ -0,0 +1,160 @@
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import java.io.File
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class ImageCaptureDeviceTest {
+
+    @get:Rule
+    val cameraPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(
+            android.Manifest.permission.CAMERA,
+            android.Manifest.permission.READ_EXTERNAL_STORAGE,
+            android.Manifest.permission.WRITE_EXTERNAL_STORAGE
+        )
+
+    private val instrumentation = InstrumentationRegistry.getInstrumentation()
+    private var activityScenario: ActivityScenario<MainActivity>? = null
+    private val uiDevice = UiDevice.getInstance(instrumentation)
+
+    @Test
+    fun image_capture_default() = runTest {
+        val timeStamp = System.currentTimeMillis()
+        var directory = File(DIR_PATH)
+        val files = directory.listFiles()
+        activityScenario = ActivityScenario.launch(MainActivity::class.java)
+        uiDevice.wait(
+            Until.findObject(By.res("CaptureButton")),
+            5000
+        )
+        uiDevice.findObject(By.res("CaptureButton")).click()
+        uiDevice.wait(
+            Until.findObject(By.res("ImageCaptureSuccessToast")),
+            5000
+        )
+        val pictureTaken = (files.size + 1) == directory.listFiles().size
+        assert(pictureTaken)
+        if (pictureTaken) {
+            deleteFilesInDirAfterTimestamp(timeStamp)
+        }
+    }
+
+    @Test
+    fun image_capture_external() = runTest {
+        val timeStamp = System.currentTimeMillis()
+        var directory = File(DIR_PATH)
+        val files = directory.listFiles()
+        val launchIntent = Intent()
+        launchIntent.setAction(MediaStore.ACTION_IMAGE_CAPTURE)
+        launchIntent.putExtra(MediaStore.EXTRA_OUTPUT, TEST_URI)
+        launchIntent.setComponent(
+            ComponentName("com.google.jetpackcamera", "com.google.jetpackcamera.MainActivity")
+        )
+        activityScenario = ActivityScenario.launchActivityForResult(launchIntent)
+        uiDevice.wait(
+            Until.findObject(By.res("CaptureButton")),
+            5000
+        )
+        uiDevice.findObject(By.res("CaptureButton")).click()
+        uiDevice.wait(
+            Until.findObject(By.res("ImageCaptureSuccessToast")),
+            5000
+        )
+        val result = activityScenario!!.result.resultCode
+        assert(result == Activity.RESULT_OK)
+        val pictureTaken = (files.size + 1) == directory.listFiles().size
+        assert(pictureTaken)
+        if (pictureTaken) {
+            deleteFilesInDirAfterTimestamp(timeStamp)
+        }
+    }
+
+    @Test
+    fun image_capture_external_no_uri() = runTest {
```

Comment:
According to the [contract](https://developer.android.com/reference/android/provider/MediaStore#ACTION_IMAGE_CAPTURE), in this case "a small sized image is returned as a Bitmap object in the extra field".

At a minimum, you should create a bug/TODO to implement that in a follow-up PR.